### ISSUE TITLE
[arcface_torch] questions about more design purpose for learning rate schedule strategy . 

### ISSUE THREAD

**Issue by leoluopy** [deepinsight/insightface#1472]
hi , @anxiangsir 
i have reproduced your work , by training glint1600W dataset , and it achieved a good result. However , i have some question about the strategy you use for the learning rate schedule . 
as we can see from the code.

```
...
config.batch_size = 64
config.lr = 0.1  # batch size is 512
...
...
    opt_pfc = torch.optim.SGD(
        params=[{'params': module_partial_fc.parameters()}],
        lr=cfg.lr / 512 * cfg.batch_size * world_size,
        momentum=0.9, weight_decay=cfg.weight_decay)
...
...
def lr_step_func(epoch):
        return ((epoch + 1) / (4 + 1)) ** 2 if epoch < config.warmup_epoch else 0.1 ** len(
            [m for m in [8, 12, 15, 18] if m - 1 <= epoch])
...
...
    scheduler_pfc = torch.optim.lr_scheduler.LambdaLR(
        optimizer=opt_pfc, lr_lambda=cfg.lr_func)
...
```
the values for cfg.lr and optimizer.param_groups['lr] will be like below:

- epoch                                                0~6
- cfg.lr                                                  1.0
- optimizer.param_groups['lr]               0.025

- epoch                                                7~11
- cfg.lr                                                  0.1
- optimizer.param_groups['lr]               0.0025

- epoch                                                12~14
- cfg.lr                                                  0.01
- optimizer.param_groups['lr]               0.00025

and finally my question comes:
       why do you use " lr=cfg.lr / 512 * cfg.batch_size * world_size " , why do you divide 512 and multiply batch_size , multiply world_size later ? 
       any learning rate schedule theory basis for distributed training  ? 
       is my question properly described ?
       waiting for your reply, thanks .

**Comment by anxiangsir** [comment]
The learning rate needs to increase linearly according to batchsize, at the same time it should apply a longer warmup phase for training.
We refer to this:
Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour
